lib/identity_cache/cached/attribute_by_multi.rb

```diff
@@ -84,7 +84,7 @@ def load_multi_rows_query(keys)
           # This results in a single "WHERE field IN (values)" statement being
           # produced from a single query.
           field_idx = other_field_indexes.first
-          field_name = key_fields[i]
```

Comment:
Oh, I didn't realize this code path wasn't tested.  This does indeed lead to a undefined local variable error with the regression test and without this change.

Reply:
Yeah, this is something I thought we caught in [this test](https://github.com/Shopify/identity_cache/blob/4bcb03096dfe5bf9f083e3bcff494039016dc140/test/fetch_multi_by_test.rb#L119-L126) but it's not structured in such a way as to hit the path -- would be nice to find a non-hacky way of running a `query.to_sql`-like check to _ensure_ all these paths are hit in case parts of this suite are restructured in the future